### PR TITLE
Add soap-to-rest and API import to work with registry abstract layer

### PR DESCRIPTION
**Pull request by chamilaadhi**
Fix https://github.com/wso2/product-apim/issues/9827

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIImportUtil.java [hunk 1]
```diff
@@ -413,23 +424,18 @@ private static void addAPISequences(String pathToArchive, API importedApi, Regis
      *
      * @param pathToArchive location of the extracted folder of the API
      * @param importedApi   the imported API object
+     * @param apiProvider 
+     * @throws APIManagementException 
      */
-    private static void addAPISpecificSequences(String pathToArchive, API importedApi, Registry registry) {
+    private static void addAPISpecificSequences(String pathToArchive, API importedApi, APIProvider apiProvider, String orgId) throws APIManagementException {
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIImportUtil.java:430]: Is this exceeding the character limit?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIImportUtil.java [hunk 2]
```diff
@@ -665,11 +691,12 @@ private static void addSOAPToREST(String pathToArchive, API importedApi, Registr
             try {
                 // Import inflow mediation logic
                 Path inFlowDirectory = Paths.get(inFlowFileLocation);
-                ImportMediationLogic(inFlowDirectory, registry, soapToRestLocationIn);
-
+                //ImportMediationLogic(inFlowDirectory, registry, soapToRestLocationIn);
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIImportUtil.java:694]: Shall we remove the commented code

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java [hunk 3]
```diff
@@ -3273,12 +3273,13 @@ private API importSOAPToRESTAPI(InputStream fileInputStream, Attachment fileDeta
             String wsdlArchiveExtractedPath, API apiToAdd) throws APIManagementException {
         try {
             APIProvider apiProvider = RestApiCommonUtil.getLoggedInUserProvider();
+            String tenantDomain = RestApiCommonUtil.getLoggedInUserTenantDomain();
             //adding the api
-            apiProvider.addAPI(apiToAdd);
+            API createdApi = apiProvider.addAPI(apiToAdd);
 
-            APIIdentifier createdApiId = apiToAdd.getId();
+            //APIIdentifier createdApiId = apiToAdd.getId();
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java:3280]: Commented code

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java [hunk 4]
```diff
@@ -3273,12 +3273,13 @@ private API importSOAPToRESTAPI(InputStream fileInputStream, Attachment fileDeta
...
             //Retrieve the newly added API to send in the response payload
-            API createdApi = apiProvider.getAPI(createdApiId);
+            //API createdApi = apiProvider.getAPI(createdApiId);
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java:3282]: Commented code

Comment (chamilaadhi) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java]: fixed

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java (hunk 3 above)
Comment (chamilaadhi) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java]: fixed

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIImportUtil.java (hunk 2 above)
Comment (chamilaadhi) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIImportUtil.java]: removed

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIImportUtil.java (hunk 1 above)
Comment (chamilaadhi) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIImportUtil.java]: fixed

**Review by mushthaq33 — APPROVED** [review]

**Pull request merged**